src/Compilers/CSharp/Portable/Operations/CSharpOperationNodes.cs

```diff
@@ -1501,24 +1504,53 @@ public CSharpLazyRecursivePatternOperation(CSharpOperationFactory operationFacto
                    syntax: boundRecursivePattern.Syntax,
                    isImplicit: boundRecursivePattern.WasCompilerGenerated)
         {
-            this._operationFactory = operationFactory;
-            this._boundRecursivePattern = boundRecursivePattern;
+            _operationFactory = operationFactory;
+            _boundRecursivePattern = boundRecursivePattern;
 
         }
         public override ImmutableArray<IPatternOperation> CreateDeconstructionSubpatterns()
         {
             return _boundRecursivePattern.Deconstruction.IsDefault ? ImmutableArray<IPatternOperation>.Empty :
                 _boundRecursivePattern.Deconstruction.SelectAsArray((p, fac) => (IPatternOperation)fac.Create(p.Pattern), _operationFactory);
         }
-        public override ImmutableArray<(ISymbol, IPatternOperation)> CreatePropertySubpatterns()
+        public override ImmutableArray<IPropertySubpatternOperation> CreatePropertySubpatterns()
+        {
+            return _boundRecursivePattern.Properties.IsDefault ? ImmutableArray<IPropertySubpatternOperation>.Empty :
+                _boundRecursivePattern.Properties.SelectAsArray((p, fac) => fac.CreatePropertySubpattern(p, MatchedType), _operationFactory);
```

Comment:
>MatchedType [](start = 108, length = 11)

Is this going to prevent the lambda from caching the delegate? Consider passing it in a tuple parameter, together with operation factory.